inspirehep/modules/records/es_record.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2016 CERN.
+#
+# INSPIRE is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Record subclass to fetch records from ElasticSearch."""
+
+from __future__ import absolute_import, division, print_function
+
+import arrow
+
+from elasticsearch.exceptions import NotFoundError
+
+from invenio_pidstore.errors import PIDDoesNotExistError
+from invenio_records.api import Record
+
+from inspirehep.utils.record_getter import (
+    RecordGetterError,
+    get_es_record_by_uuid
+)
+
+
+class ESRecord(Record):
+
+    """Record class that fetches records from ElasticSearch."""
+
+    @classmethod
+    def get_record(cls, object_uuid, with_deleted=False):
+        """Get record instance from ElasticSearch."""
+        try:
+            return cls(get_es_record_by_uuid(object_uuid))
+        except RecordGetterError as e:
+            if isinstance(e.cause, NotFoundError):
+                # Raise this error so the interface will render a 404 page
+                # rather than a 500
+                raise PIDDoesNotExistError('es_record', object_uuid)
```

Comment:
yep, in theory it should be `pidtype` and `pidvalue` (which we don't have here) so we are 'abusing' the same error in this case. This is because `invenio-records-ui` catches it and shows 404.